otherlibs/systhreads/thread.ml

```diff
@@ -85,9 +85,9 @@ let preempt_signal =
   | _       -> Sys.sigvtalrm
 
 let () =
-  Sys.set_signal preempt_signal (Sys.Signal_handle preempt);
-  Domain.at_startup thread_initialize_domain;
   thread_initialize ();
+  Domain.at_startup thread_initialize_domain;
```

Comment:
Is there a reason for switching the order of these functions?

Reply:
Yes, thread_initialize can now raise an exception so we do it before any other changes. Since after the call you are sure that only one domain runs, this does not change behaviour.

Reply:
Thanks.